org.mixedrealitytoolkit.input/Readers/PinchInputReader.cs

```diff
@@ -145,16 +141,14 @@ protected virtual void OnDisable()
         /// </summary>
         protected virtual void Update()
         {
-            if (m_firstUpdate)
-            {
-                UpdateActionValidCaches();
-                m_firstUpdate = false;                
-            }
+            m_isTrackingStatePolyfilled = trackedPoseDriver.GetIsPolyfillDevicePose();
```

Comment:
Do we want to require a trackedPoseDriver?  In cases where the app can rely totally on action mappings, it doesn't need to pass in a traackedPoseDriver here.

I would prefer to check for null here.

Reply:
For completeness, I double-checked that setting `trackedPoseDriver` to null is safe to do here. It is! The extension method correctly returns `false` here. So, I believe the proposed scenario (not needing to pass in a traackedPoseDriver) is supported.

Reply:
My view is if we are now checking the validity of the select and select value actions every frame, not sure we need to also check if the tracked pose is polyfilled as well.  If either select action is not valid in a given frame, we fallback to polyfilled selection - what purpose does checking the tracked pose serve at that point?

Reply:
> If either select action is not valid in a given frame, we fallback to polyfilled selection - what purpose does checking the tracked pose serve at that point?

There exist some platforms (like Snapdragon Spaces) that don't clear a controller interaction profile even when controllers are disconnected (iirc, Quest _does_, but either path is within spec):

> The runtime **may** return the last-known interaction profile in the event that no controllers are active.

In this case, assuming the Hand Interaction Profile hasn't been added to the app's settings for whatever reason (like the way our example project is _currently_ set up, but I'll update that soon 😉), when controllers are disconnected and the user switches to hands, the select actions will still have bound controls (since the controller interaction profile is still reporting active), but the tracking state will properly relay the untracked nature of the controllers, allowing us to fall back.

My original change to support this case in XRI2 can be found at https://github.com/MixedRealityToolkit/MixedRealityToolkit-Unity/pull/704, for more info.

Reply:
@keveleigh Ah, I see.  Thanks for the explanation!  I see now where that was introduced (and why) and how this was just replacing that functionality.